src/main/java/Data/StoreList.java

```diff
@@ -0,0 +1,209 @@
+package Data;
+
+import Exceptions.*;
+import Tasks.*;
+
+import java.util.ArrayList;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDate;
+
+public class StoreList {
+
+    //declare array to store tasks
+    protected ArrayList<Task> items;
+
+    //declare Tasks.Task
+    protected Task t;
+
+    //initialize items array
+    public StoreList(ArrayList<Task> items) {
+        this.items = items;
+    }
+
+    public ArrayList<Task> getItems() {
+        return items;
+    }
+
+    public Task get(int i) {
+        return items.get(i);
+    }
+
+    /**
+     * Adds appropriate task to list
+     * Catch all the relevant exceptions when trying to add invalid tasks
+     *
+     * @param item task to be added.
+     * @param type tasktype to be created and stores.
+     * @return
+     */
+    public String addItem(String item, String type) {
+        try {
+            if (type.equals("todo")) {
+                // create a Tasks.ToDos object
+                t = new ToDos(item);
+                items.add(t);
+
+            } else if (type.equals("deadline")) {
+                // create a Tasks.Deadlines object
+                t = new Deadlines(item); // Constructor might throw Exceptions.EmptyDeadlineException
+                items.add(t);
+
+            } else {
+                // create an Tasks.Events object
+                t = new Events(item); // Constructor might throw Exceptions.EmptyEventException
+                items.add(t);
+            }
+
+            return "    Got it. I've added this task:\n" + "      " + t.print() +
+                    "\n    Now you have " + this.getSize() + " tasks in the list.";
+
+        } catch (EmptyDescException | EmptyDeadlineException
+                 | EmptyEventException | EmptyDeadlineDateException
+                 | EmptyEventTimingException e) {
+
+            return e.getMessage();
+
+        } catch (Exception e) {
+            return "An unexpected error occurred: " + e.getMessage();
+        }
+    }
+
+    //getter
+    public int getSize() {
+
+        return items.size();
+    }
+
+    /**
+     * Marks item as completed
+     *
+     * @param num Index of task to be marked.
+     */
+    public String markItem(int num) throws InvalidIndexException {
+        if (num > items.size()) {
+            throw new InvalidIndexException("Task number does not exist");
+        }
+        items.get(num - 1).mark();
+        return "    Wohoo! I've marked this task as done! WELL DONE!:\n" +
+                "      " + items.get(num - 1).print();
+    }
+
+    /**
+     * Unmarks item as incomplete
+     *
+     * @param num Index of task to be unmarked.
+     */
+    public String UnmarkItem(int num) throws InvalidIndexException {
```

Comment:
Wouldn't it be better for following the coding standard if the method name is unmarkItem ?